Src/Base/AMReX_Random.H

```
@@ -37,14 +87,20 @@ namespace amrex
     */
     AMREX_GPU_EXTERNAL AMREX_GPU_HOST_DEVICE unsigned int RandomPoisson (Real lambda);
 
-
-
-#ifdef AMREX_USE_GPU
-    // Locking mechanism functions for locking and unlocking
-    AMREX_GPU_DEVICE int get_state (int tid);
-    AMREX_GPU_DEVICE void free_state (int tid);
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    unsigned int RandomPoisson (Real lambda, RandomEngine const& random_engine)
+    {
+#if defined(__CUDA_ARCH__)
+        return curand_poisson(random_engine.rand_state, lambda);
+#elif defined(__HIP_DEVICE_COMPILE__)
+        return hiprand_poisson(random_engine.rand_state, lambda);
+#elif defined (__SYCL_DEVICE_ONLY__)
+        return 0; // xxxxx DPCPP todo: Poisson distribution
```

maybe these should explicitly fail instead of returning 0